src/contrib/php_array_api.h

```diff
@@ -420,7 +420,17 @@ char *php_array_zval_to_string(zval *z, int *plen, zend_bool *pfree) {
  * zval *php_array_fetchz_array(zval *zarr, zval *key)
  */
 static inline zval *php_array_zval_to_array(zval *zarr) {
-	return (zarr && (Z_TYPE_P(zarr) == IS_ARRAY)) ? zarr : NULL;
+try_again:
+	if (!zarr) { return NULL; }
+	switch (Z_TYPE_P(zarr)) {
+		case IS_REFERENCE:
+			ZVAL_DEREF(zarr);
+			goto try_again;
+		case IS_ARRAY:
+			return zarr;
+		default:
+			return NULL;
+	}
```

Comment:
This looks related to a previous issue we had with dereferencing strings. See https://github.com/mongodb/mongo-php-driver/commit/c68cdcda7654a630b6da52d99f7ce5e2b3fdabfc from [PHPC-1839](https://jira.mongodb.org/browse/PHPC-1839).

Also, you should contribute all of these `ZVAL_DEREF()` changes upstream to [sgolemon/php-array-api](https://github.com/sgolemon/php-array-api/) when you get a chance. It's your ticket to the very exclusive [contributors club](https://github.com/sgolemon/php-array-api/graphs/contributors).

Maybe @sgolemon can make us t-shirts.

Reply:
Will contribute this upstream 👍